tests/testthat/test-layer_predict.R

```diff
@@ -31,3 +31,47 @@ test_that("prediction with interval works", {
   expect_equal(nrow(p), 108L)
   expect_named(p, c("geo_value", "time_value", ".pred_lower", ".pred_upper"))
 })
+
+test_that("layer_predict dots validation", {
+  # We balk at unnamed arguments, though perhaps not with the most helpful error messages:
+  expect_error(
+    frosting() %>% layer_predict("pred_int", list(), tibble::tibble(x = 5)),
+    class = "epipredict__layer_predict__unnamed_dot"
+  )
+  expect_error(
+    frosting() %>% layer_predict("pred_int", list(), "maybe_meant_to_be_id"),
+    class = "epipredict__layer_predict__unnamed_dot"
+  )
+  # We allow arguments that might actually work at prediction time:
+  expect_no_error(frosting() %>% layer_predict(type = "quantile", interval = "confidence"))
+
+  # We don't detect completely-bogus arg names until predict time:
+  expect_no_error(f_bad_arg <- frosting() %>% layer_predict(bogus_argument = "something"))
+  wf_bad_arg <- wf %>% add_frosting(f_bad_arg)
+  expect_error(predict(wf_bad_arg, latest))
+  # Some argument names only apply for some prediction `type`s; we don't check for ignored arguments, and neither does workflows:
+  expect_no_error(frosting() %>% layer_predict(eval_time = "preferably this would error"))
+
+  # ^ (currently with a truly awful error message, due to an extra comma in parsnip::check_pred_type_dots)
+  #
+  # Unfortunately, we outright ignore attempts to pass args via `predict.epi_workflow`:
+  f_predict <- frosting() %>% layer_predict()
+  wf_predict <- wf %>% add_frosting(f_predict)
+  expect_no_error(predict(wf_predict, latest, type = "pred_int"))
+})
+
+test_that("layer_predict dots are forwarded", {
+  f_lm_int_level <- frosting() %>%
+    layer_predict(type = "pred_int", level = 0.8)
+  wf_lm_int_level <- wf %>% add_frosting(f_lm_int_level)
+  p <- predict(wf, latest)
+  p_lm_int_level <- predict(wf_lm_int_level, latest)
+  expect_contains(names(p_lm_int_level), c(".pred_lower", ".pred_upper"))
+  expect_equal(nrow(na.omit(p)), nrow(na.omit(p_lm_int_level)))
+  expect_true(cbind(p, p_lm_int_level[c(".pred_lower", ".pred_upper")]) %>%
+    na.omit() %>%
+    mutate(sandwiched = .pred_lower <= .pred & .pred <= .pred_upper) %>%
+    `[[`("sandwiched") %>%
+    all())
+  # There are many possible other valid configurations that aren't tested here.
```

Comment:
One potential further issue (possibly out of scope) is the one RyanN uncovered: if you manage to get your workflow to fit multiple response variables simultaneously, the `type = "pred_int"` chokes. 

``` r
library(workflows)
library(recipes)

tib <- tibble(y1 = rnorm(100), y2 = rnorm(100), x = rnorm(100), z = rnorm(100))
rec <- recipe(y1 + y2 ~ x + z, data = tib) %>%
  step_normalize(all_numeric_predictors())
wf <- workflow(rec, parsnip::linear_reg()) %>% generics::fit(tib)
predict(wf, tib[1:5, ])
#> # A tibble: 5 × 2
#>   .pred_y1 .pred_y2
#>      <dbl>    <dbl>
#> 1    0.285 -0.0736 
#> 2   -0.102 -0.0145 
#> 3   -0.206 -0.00517
#> 4   -0.180 -0.00994
#> 5   -0.256  0.0434
predict(wf, tib[1:5, ], type = "pred_int")
#> Error in `dplyr::select()`:
#> ! Can't subset columns that don't exist.
#> ✖ Column `fit` doesn't exist.
```

Reply:
Maybe related, maybe not: I just remembered that you can get `predict(ewf, dat, type = "pred_int")` to work in some cases... but I haven't made `predict(ewf, dat, type = "pred_int", level = 0.8)` work.  Do we want to make the former an error, or the latter work?

Reply:
That seems like an upstream bug or bugs that we also trip over....
- parsnip either generates an interval only for y2, or generates this error, depending on whether you write `y1 + y2 ~` or `cbind(y1, y2) ~`
- recipes tries to prevent you from writing `cbind(y1, y2) ~`
- workflows transforms a recipe formula from `y1 + y2 ~` to `cbind(y1, y2) ~`

We could attempt to do better here by trying to detect & either abort or patch over these bad cases... but should we just be filing/fixing upstream?